### PR TITLE
More warning fixes

### PR DESCRIPTION
**Pull request by ZedThree**
Fix lots more warnings

### CODE REVIEW

File: include/cyclic_reduction.hxx [hunk 1]
```diff
@@ -343,6 +343,7 @@ public:
     ///////////////////////////////////////
     // Solve local equations
     back_solve(Nsys, N, coefs, x1, xn, x);
+    delete req;
```
Comment (dschwoerer) [inline include/cyclic_reduction.hxx]: This causes undefined behaviour! Use delete[]

File: src/field/globalfield.cxx [hunk 2]
```diff
@@ -164,6 +164,7 @@ void GlobalField2D::gather(const Field2D &f) {
         }
       }while(pe != MPI_UNDEFINED);
     }
+    delete req;
```
Comment (dschwoerer) [inline src/field/globalfield.cxx]: use `delete[]`

File: src/field/globalfield.cxx [hunk 3]
```diff
@@ -329,6 +330,7 @@ void GlobalField3D::gather(const Field3D &f) {
...
```
Comment (dschwoerer) [inline src/field/globalfield.cxx]: delete[]

File: src/mesh/data/gridfromoptions.cxx [hunk 4]
```diff
@@ -91,45 +91,45 @@ bool GridFromOptions::get(Mesh *m, vector<int> &var, const string &name, int len
...
+
   // Parse, and evaluate with x,y,z,t = 0
   FieldGenerator* gen = FieldFactory::get()->parse(expr, options);
 
   var.resize(len);
-  
+
   switch(dir) {
   case GridDataSource::X: {
-    for(int x=0;x<var.size();x++){
```
Comment (dschwoerer) [inline src/mesh/data/gridfromoptions.cxx]: I assume the compiler warnung was comparison of signed and unsigned int?
Wouldn't it make more sense to use an unsigned int for `x` and `len`?

File: include/cyclic_reduction.hxx (hunk 1 above)
Comment (ZedThree) [inline include/cyclic_reduction.hxx]: Good catch, thanks!

**Pull request merged**